rfd/00xx-snowflake-support.md

```diff
@@ -0,0 +1,170 @@
+---
+authors: Jakub Nyckowski (jakub.nyckowski@goteleport.com)
+state: draft
+---
+
+# RFD XX - Snowflake Database support
+
+## What
+
+[Snowflake](https://www.snowflake.com/) integration in Teleport Database Access. 
+
+## Why
+
+We want to increase the number of databases supported by Teleport. 
+
+## Details
+
+The Snowflake support can be split into 4 parts:
+
+1. Database access support - changes inside Teleport Data Access.
+2. CLI support - `snowsql` the official Snowflake client.
+3. GUI clients support - DataGrip by JetBrains, DBeaver and similar.
+4. WebUI integration https://app.snowflake.com/.
+
+### Snowflake architecture
+
+* Snowflake uses REST API for communication. 
+* Payload is a regular JSON. Some queries can also carry payload in [Arrow format](https://arrow.apache.org/). 
+* Snowflake uses a slightly modified JWT token flow for authentication (details below).
+
+### Teleport configuration
+
+Like in case of other databases, a user will have to add an entry to the teleport configuration file. Example:
+
+```yaml
+db_service:
+  enabled: "yes"
+  databases:
+      - name: "snowflake"
+      description: "Snowflake database"
+      protocol: "snowflake"
+      uri: "im12345.us-east-2.aws.snowflakecomputing.com"
+```
+
+The next step would be to add teleport CA to the Snowflake user account. `tctl` could learn to export the required keys 
```

Comment:
Are we going to use our new Database CA to sign JWT tokens?

Reply:
I think so. We don't need to port the integration to v9 and any other CA at that point will be confusing as all other databases are using the Database CA. Rotation could potentially break the integration.